rs/sns/testing/src/utils.rs

```diff
@@ -0,0 +1,245 @@
+use anyhow::{anyhow, Context, Result};
+use dfx_core::config::model::network_descriptor::NetworkDescriptor;
+use dfx_core::identity::identity_manager::InitializeIdentity;
+use dfx_core::identity::IdentityManager;
+use futures::future::join_all;
+use ic_agent::Identity;
+use ic_agent::{
+    agent::route_provider::RoundRobinRouteProvider, identity::Secp256k1Identity, Agent,
+};
+use ic_base_types::CanisterId;
+use ic_base_types::PrincipalId;
+use ic_nervous_system_agent::nns::governance::list_neurons;
+use ic_nervous_system_agent::nns::sns_wasm::{get_latest_sns_version_pretty, get_sns_subnet_ids};
+use ic_nervous_system_agent::CallCanisters;
+use ic_nervous_system_common_test_keys::TEST_NEURON_1_ID;
+use ic_nns_common::pb::v1::NeuronId;
+use ic_nns_constants::{
+    canister_id_to_nns_canister_name, CYCLES_LEDGER_CANISTER_ID, CYCLES_MINTING_CANISTER_ID,
+    GOVERNANCE_CANISTER_ID, LEDGER_CANISTER_ID, LEDGER_INDEX_CANISTER_ID, LIFELINE_CANISTER_ID,
+    REGISTRY_CANISTER_ID, ROOT_CANISTER_ID, SNS_WASM_CANISTER_ID,
+};
+use ic_nns_governance_api::pb::v1::{ListNeurons, Neuron};
+use k256::SecretKey;
+use lazy_static::lazy_static;
+use reqwest::Client;
+use thiserror::Error;
+
+pub const ALL_NNS_CANISTER_IDS: [&CanisterId; 9] = [
+    &GOVERNANCE_CANISTER_ID,
+    &LEDGER_CANISTER_ID,
+    &ROOT_CANISTER_ID,
+    &LIFELINE_CANISTER_ID,
+    &SNS_WASM_CANISTER_ID,
+    &REGISTRY_CANISTER_ID,
+    &CYCLES_MINTING_CANISTER_ID,
+    &LEDGER_INDEX_CANISTER_ID,
+    &CYCLES_LEDGER_CANISTER_ID,
+];
+
+pub const NNS_NEURON_ID: NeuronId = NeuronId {
+    id: TEST_NEURON_1_ID,
+};
+
+// Predefined secret keys used in sns-testing
+lazy_static! {
+    pub static ref TREASURY_SECRET_KEY: SecretKey = {
+        let mut slice_vec = vec![0; 16];
+        slice_vec.extend_from_slice(&200_usize.to_ne_bytes());
+        SecretKey::from_slice(&slice_vec).unwrap()
+    };
+    pub static ref TREASURY_PRINCIPAL_ID: PrincipalId = {
+        let identity = Secp256k1Identity::from_private_key(TREASURY_SECRET_KEY.clone());
+        identity.sender().unwrap().into()
+    };
+}
+
+pub fn swap_participant_secret_keys(number_of_participants: usize) -> Vec<SecretKey> {
+    (0..number_of_participants)
+        .map(|i| {
+            let mut slice_vec = vec![0; 16];
+            slice_vec.extend_from_slice(&(100 + i).to_ne_bytes());
+            SecretKey::from_slice(&slice_vec).unwrap()
+        })
+        .collect()
+}
+
+pub async fn build_ephemeral_agent(
```

Comment:
what does ephemeral mean in this context?

Reply:
That it doesn't have a corresponding `dfx` identity, but perhaps the name could've been chosen better

Reply:
Maybe just add a comment explaining.  I can't think of a better name, I just didn't understand what it was for.